modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,344 @@
+Copyright (C) 2010-2013, Advanced Micro Devices, Inc., all rights reserved.
+Copyright (C) 2015, OpenCV Foundation, all rights reserved.
+Copyright (C) 2015, Itseez Inc., all rights reserved.
+Third party copyrights are property of their respective owners.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+  * Redistributions of source code must retain the above copyright notice,
+    this list of conditions and the following disclaimer.
+
+  * Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+
+  * Neither the names of the copyright holders nor the names of the contributors
+    may be used to endorse or promote products derived from this software
+    without specific prior written permission.
+
+This software is provided by the copyright holders and contributors "as is" and
+any express or implied warranties, including, but not limited to, the implied
+warranties of merchantability and fitness for a particular purpose are disclaimed.
+In no event shall copyright holders or contributors be liable for any direct,
+indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.
+*/
+#include "opencv2/face/face_alignment.hpp"
+#include "opencv2/core/mat.hpp"
+#include <opencv2/objdetect.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/core.hpp>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <map>
+#include <vector>
+#include <cmath>
+#include <vector>
+using namespace std;
+namespace cv{
+namespace face{
+const std::string face_cascade_name = "lbpcascade_frontalface_improved.xml";
+const std::string meanshapefile ="opencv2/modules/face/data/mean_shape.txt";
+class ShapePredictor : public cv::face::FacemarkKazemi{
+public:
+    virtual int getNumLandmarks() const {return numlandmarks_;}
+    virtual int getNumFaces() const {return numfaces_;}
+    ShapePredictor(){
+        //get name  of file containing meanshape
+        ifstream f(meanshapefile.c_str());
+        string s;
+        stringstream ss;
+        //read value of annotations
+        while(getline(f,s)){
+            unsigned long i=0;
+            char x1[100],y1[100];
+            while(s[i]!=','){
+                x1[i]=s[i];
+                i++;
+            }
+            string x,y;
+            x=string(x1);
+            Point2f pt;
+            ss<<x;
+            ss>>pt.x;
+            ss.clear();
```

Comment:
thats because you should use [atof](http://www.cplusplus.com/reference/cstdlib/atof/) instead of stof. That works on every platform to convert a string to double. Should work just fine.